src/auth/http-poller-with-promises.ts

```diff
@@ -0,0 +1,51 @@
+import to from 'await-to-js';
+
+import createHttpPoller, { HttpPoller } from './http-poller';
+
+export type RetryingPollerProps = {
+  pollFunction: () => Promise<Response | undefined>;
+  delayInMs?: number;
+  maxRetries?: number;
+};
+
+export default async function retryPollingUntilSuccessful({
+  pollFunction,
+  delayInMs = 1000,
```

Comment:
http-poller uses the term `pollIntervalInMs` for this. It could be named the same here.